File: Sorting Algorithms/InsertionSort.py
__author__='Kalpak Seal'

def insertion_sort(arr):
    n = len(arr)

    for i in range(1, n):
        key = arr[i]

        j = i-1
        while j >= 0 and key < arr[j]:
            arr[j+1] = arr[j]
            j -= 1

        arr[j+1] = key
    return arr

if __name__== '__main__':
    arr = [int(x) for x in input("Enter the list of numbers to be sorted").split()]
    print(insertion_sort(arr))

File: Sorting Algorithms/Algorithms/InsertionSort.java
public class InsertionSort
{
    public static void sort(int[] arr)
    {
        for(int j = 1; j < arr.length; j++)
        {
            int key = arr[j];
            int i = j - 1;

            while(i >= 0 && arr[i] > key)
            {
                arr[i+1] = arr[i];
                i -= 1;
            }
            arr[i+1] = key;
        }
    }

    public static void main(String[] args)
    {
        int[] arr = {5, 2, 4, 6, 1, 3};
        sort(arr);

        for(int i : arr)
            System.out.print(i + " ");
        System.out.println();
    }
}